### PR TITLE
Fewer test warnings in 3.7

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description
[Tell us about your new feature, improvement, or fix! If relevant, please supplement the PR with images.]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by jni** [comment]
@hmaarrfk do you know why all of Travis is red?

**Comment by hmaarrfk** [comment]
i feel like someby else just reported this issue

**Comment by hmaarrfk** [comment]
The pip issue cross referenced here https://github.com/scikit-image/scikit-image/issues/3685 states that this is an issue arises with the new pip and `pyproject.toml`. Either we aren't using pyproject.toml correctly or pip made a mistake. I think it is best if we just remove pyproject.toml.

xref: https://github.com/scikit-image/scikit-image/issues/3576

cc: @stefanv

**Comment by hmaarrfk** [comment]
I moved the removal of the pyproject.toml to https://github.com/scikit-image/scikit-image/issues/3576

File: skimage/exposure/tests/test_exposure.py [hunk 1]
```diff
@@ -276,7 +276,8 @@ def test_adapthist_grayscale():
     img = skimage.img_as_float(data.astronaut())
     img = rgb2gray(img)
     img = np.dstack((img, img, img))
-    with expected_warnings(['precision loss|non-contiguous input']):
+    with expected_warnings(['precision loss|non-contiguous input',
+                            r'divide by zero|\A\Z']):
```
Comment (stefanv) [inline skimage/exposure/tests/test_exposure.py]: Err, that doesn't look right.  We should not be dividing by zero!

File: skimage/graph/tests/test_heap.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def _test_heap(n, fast_update):
     a = [random.uniform(1.0, 100.0) for i in range(n // 2)]
     a = a + a
 
-    t0 = time.clock()
+    t0 = time.perf_counter()
```
Comment (stefanv) [inline skimage/graph/tests/test_heap.py]: What's changed here?

File: skimage/exposure/tests/test_exposure.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/exposure/tests/test_exposure.py]: haha good point!

File: skimage/graph/tests/test_heap.py (hunk 2 above)
Comment (hmaarrfk) [reply skimage/graph/tests/test_heap.py]: apparently `time.clock` is just deprecated.

```
$ python --version                                                                                                                                                                                          
Python 3.7.1
mark2@xps ~                                                                                                                                                                                                 
$ python -c 'import time; time.clock()'                                                                                                                                                                     
-c:1: DeprecationWarning: time.clock has been deprecated in Python 3.3 and will be removed from Python 3.8: use time.perf_counter or time.process_time instead
```

File: skimage/exposure/tests/test_exposure.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/exposure/tests/test_exposure.py]: I removed this change allowing this PR to go forward. Somebody will have to investigate this warning.

**Comment by sciunto** [comment]
We might want to backport to 0.14.3, especially if it comes before 0.15. @hmaarrfk Would you mind backporting it if it seems relevant for you?

**Comment by hmaarrfk** [comment]
I think somebody would have to touch up collections.abc with a try/catch statement

@meeseeksdev backport to v0.14.x

**Pull request merged**